src/network.rs

```diff
@@ -248,7 +281,10 @@ pub fn serve(port: u16) {
             match event.token() {
                 SOCK => {
                     let (len, addr) = sockfd.recv_from(&mut buf).unwrap();
-                    let msg: Message = deserialize(&buf[0..len]).unwrap();
+                    //info!("{} {:x} {:x}", len, buf[0], buf[1]);
+                    let mut decrypted_buf = aead::open_in_place(&opening_key, &nonce, &additional_data, 0, &mut buf).unwrap();
```

Comment:
1. `decrypted_buf` doesn't need to be mutable.
2. `buf` is a fixed-length buffer, so instead of passing `buf` to  `open_in_place`, you should use a slice of buffer: `&mut buf[0..len]`